cedars/app/auth.py

```diff
@@ -133,6 +170,48 @@ def login():
     return render_template('auth/login.html',  **db.get_info())
 
 
+@bp.route("/token-login", methods=["POST"])
+def token_login():
+    """Login a user using the bearer token received from the frontend"""
+    token = request.json.get('token')
+    user_id = request.json.get('user_id')
+    if not token:
+        return jsonify({"error": "No token provided"}), 400
+
+    project_info = db.get_info()
+    project_id = project_info["project_id"]
+
+    logger.info(f"Token: {token}")
+    logger.info(f"User ID: {user_id}")
+    logger.info(f"Project ID: {project_id}")
+
+    user_data = verify_external_token(token,
+                                      user_id=user_id,
+                                      project_id=project_id)
+    logger.info(f"User data: {user_data}")
+    if user_data:
+        username = user_data["user"].get('email')
+        existing_user = db.get_user(username)
+        if not existing_user:
+            # Create a new user with data from the external API
+            db.add_user(
+                username=username,
+                password=generate_password_hash(token),  # Store hashed token as password
+                is_admin=True if "admin" in user_data["user"].get('institution_roles') else False
+            )
+        user = User(db.get_user(username))
+        session["superbio_api_token"] = token
+        if 'project' in user_data and 'model' in user_data['project']:
+            # If we are are using any model with this login,
+            # then we start the PINES server and get it's URL
+            pines_url, is_url_from_api = load_pines_url(project_id,
+                                        superbio_api_token=token)
+            db.create_pines_info(pines_url, is_url_from_api)
```

Comment:
please make sure that the user is not waiting for the tenacity to complete retrying